src/modules/nfts/components/wizard/RoyaltiesCreators.tsx

```diff
@@ -332,14 +337,26 @@ export default function RoyaltiesCreators({
         if (creators.length >= MAX_CREATOR_LIMIT) {
           throw new Error('Max level of creators reached');
         }
-        setCreators([...creators, { address: values.addCreator, share: 0 }]);
+        const newShareSplit = 100 / (creators.length + 1);
+        setCreators([
+          ...creators.map((c) => ({ ...c, share: newShareSplit })),
+          { address: values.addCreator, share: newShareSplit },
+        ]);
         toggleCreatorField(false);
         form.resetFields(['addCreator']);
       })
       .catch((err) => {
         console.log('err is ', err);
       });
   };
+  const removeCreator = (creatorAddress: string) => {
+    const newShareSplit = 100 / (creators.length - 1) || 100;
+    setCreators(
+      creators
+        .filter((c) => c.address !== creatorAddress)
+        .map((c) => ({ ...c, share: newShareSplit }))
+    );
+  };
```

Comment:
Also curious on why creator state isn't being tracked through the ant form?

Does setFieldsValue not suffice?